src/main/java/com/zerobase/service/SignupService.java

```diff
@@ -0,0 +1,193 @@
+
+  private final VerifyService verifyService;
+
+  private final JwtAuthProvider provider;
+
+
+
+
+  public boolean isEmailExist(String email){
+    return accountUserRepository.findByEmail(email.toLowerCase(Locale.ROOT)).isPresent();
+  }
+
+  private boolean isAccountExist(String accountName) {
+    return accountRepository.findByAccountName(accountName.toLowerCase(Locale.ROOT)).isPresent();
+  }
+
+  public AccountUser signup(SignUpUserForm form){
+    return accountUserRepository.save(AccountUser.from(form));
+  }
+
+
+//----------------------------------------------------------------------------------------
+
+  private String getRandomCode(){
+    return RandomStringUtils.random(10, true, true);
+  }
+
+  private String getVerificationEmailBody(String email, String name, String code){
+    StringBuilder sb = new StringBuilder();
+    return sb.append("Hello ").append(name)
+        .append("! Please Click Link for verification. \n")
+        .append("http://localhost:8080/signUp")
+        .append("/verify/")
+        .append("user")
+        .append("?email=")
+        .append(email)
+        .append("&code=")
+        .append(code)
+        .toString();
+  }
+
+// ----------------------------------------------------------------------------------------
+
+  // 계좌 번호 생성기
+  public static String generateAccountNumber() {
+    Random random = new Random();
+
+    // 첫 번째 6자리 숫자 생성 (100000 이상 999999 이하)
+    int firstPart = 100000 + random.nextInt(900000);
+
+    // 두 번째 2자리 숫자 생성 (00 이상 99 이하)
+    int secondPart = random.nextInt(100);
+
+    // 세 번째 6자리 숫자 생성 (100000 이상 999999 이하)
+    int thirdPart = 100000 + random.nextInt(900000);
+
+    // 계좌 번호를 문자열로 조합하여 반환
+    return String.format("%d-%02d-%d", firstPart, secondPart, thirdPart);
+  }
+
+
+//----------------------------------------------------------------------------------------
+  @Transactional
+  public String userSignUp(SignUpUserForm form) {
+    // 이미 가입된 이메일이 있는 지 체크
+    if (isEmailExist(form.getEmail())) {
+      throw new AccountException(ErrorCode.ACCOUNT_ALREADY_EXIST);
+    } else {
+      AccountUser accountUser = signup(form);
+
+      String code = getRandomCode();
+
+      SendMailForm sendMailForm = SendMailForm.builder()
+          .from("zerobase-Auth@email.com")
+          .to(form.getEmail())
+          .subject("Verification Email")
+          .text(getVerificationEmailBody(form.getEmail(), form.getName(), code))
+          .build();
+
+      System.out.println("EMAIL FORM :"+sendMailForm);
+
+      mailgunClient.sendEmail(sendMailForm);
+
+      verifyService.ChangeUserVerification(accountUser.getId(), code);
+
+      return "회원 가입 성공";
+    }
+  }
+
+  @Transactional
+  public void userVerify(String email, String code) {
+    AccountUser accountUser =  accountUserRepository.findByEmail(email)
+        .orElseThrow(() -> new AccountException(ErrorCode.USER_NOT_FOUND));
+
+    if(!accountUser.getVerificationCode().equals(code)){
```

Comment:
```suggestion
    if (!accountUser.getVerificationCode().equals(code)) {
```
coding convention 에 대해서는 관련 내용을 확인해보시고, 항상 숙지해보는 습관을 들이면 좋을 것 같아요.
실제 현업에서는 개발자들간의 coding style 을 통일하여 전체적인 style 을 맞추어 혼란을 줄이려는 노력을 하고 있습니다.
https://naver.github.io/hackday-conventions-java/